DirectEmployers/MyJobs
# [PD-2982] Make the last few holdouts for tag lookup/creation case insensitive.

## What's this PR do?
Most of the places in the codebase already look up tags using `__iexact` before doing the create. This corrects the last few that didn't.

I would really be happy if there were a way to centralize all this logic so tag-case-insensitivity didn't need to be some super-secret knowledge passed from developer to developer.

## Where should the reviewer start?
It's a small change. The biggest question is if there anything I missed?

## How should this be manually tested?
Create a tag.
Try to re-create that same tag, just different case sensitivity, everywhere you can. In all instances it should use the existing tag. It shouldn't create a new tag.

## Any background context you want to provide?
The ticket calls for correcting existing tags, but I wasn't able to find any in production. There may be a following PR with a migration if someone proves me wrong about those.
